tools/flash_shelly.py

```diff
@@ -96,16 +96,45 @@
   pipe = subprocess.check_output(['pip3', 'install', 'requests'])
   import requests
 
+def get_info(host):
+  if not '.local' in host:
+    host = host + '.local'
+  try:
+    with urllib.request.urlopen(f'http://{host}/rpc/Shelly.GetInfo') as fp:
+      info = json.load(fp)
+    info['fw_type'] = 'homekit'
+    if not 'device_id' in info:
+      info['device_id'] = info['id']
+    return((info['host'], info))
+  except (urllib.error.URLError) as err:
+    if 'Errno 8' in str(err.reason):
+      logger.warning(f"{RED}Could not resolve host: {host}\n{NC}")
+      return(None)
+    try:
```

Comment:
you can reduce indentation here because you have return at the end of the previous `try` block